src/components/Map.js

```diff
@@ -23,12 +23,16 @@ const Map = ({
   center,
 }) => {
   useEffect(() => {
-    fetchNearbyPlaces(
-      mapProperties.center.lat,
-      mapProperties.center.lng
-    ).then((res) => setNearbyPlaces(res));
+    if (!pendingPromise) {
+      fetchNearbyPlaces(
+        mapProperties.center.lat,
+        mapProperties.center.lng,
+        setPendingPromise
+      ).then((res) => setNearbyPlaces(res), setPendingPromise(false));
```

Comment:
I'm confused by the comma here 🤔 

Is the intention the following?

    ).then((res) => {
        setNearbyPlaces(res);
        setPendingPromise(false);
    });

I'm honestly surprised this doesn't throw a syntax error. Aha, [I had to look it up](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then), but here's the method signature for `.then`:

    p.then(onFulfilled[, onRejected]);

So, the comma is correct syntax because `.then` takes 2 arguments, but what that means is that `setPendingPromise(false)` is only doing what you want as a lucky side effect of the fact that you're executing the function inline.

What I think would be more clear and deterministic would be to move `setPendingPromise(false)` into the `setNearbyPlaces` function. This would also allow you to simplify this line of code down to the following:

    ).then(setNearbyPlaces);

Reply:
Wow that’s actually a really good catch.